Framework/Algorithms/inc/MantidAlgorithms/Q1DWeighted.h

```diff
@@ -76,18 +76,50 @@ class MANTID_ALGORITHMS_DLL Q1DWeighted : public API::Algorithm {
   void calculate(const API::MatrixWorkspace_const_sptr &);
   void finalize(const API::MatrixWorkspace_const_sptr &);
 
+  struct wedge {
+    wedge(double innerRadius, double outerRadius, double centerX, double centerY, double angleMiddle, double angleRange)
+        : innerRadius(innerRadius), outerRadius(outerRadius), centerX(centerX), centerY(centerY),
+          angleMiddle(angleMiddle), angleRange(angleRange) {}
+    double innerRadius;
+    double outerRadius;
+    double centerX;
+    double centerY;
+    double angleMiddle;
+    double angleRange;
+
+    bool operator<(wedge &other) {
```

Comment:
This is confusing. I think you should separate operator< for sorting, which compares only the middle angle and isMirror for checking if the two are back-to-back within some tolerance. The latter can be used instead of checkIfSymetricalWedge

Reply:
But this operator < would then return a < b  as true and b < a as true too, even with a != b.
That looks very confusing to me.

Reply:
I am not sure I understand what you mean.
`bool operator<(const Wedge&w) {
   return this->angleMiddle < w.angleMiddle;
}`
This will return true if and only if angleMiddle of a is less than that of w.

Reply:
What I don't like is that this `<` does not define a total ordering ... but on the other hand I could just pass a lambda to the sort, and it will work just fine.

Reply:
by total do you mean to compare all the properties?

Reply:
yes, sorting with lambda is also fine

Reply:
No, I mean it in the mathematical way of a "total ordering"' ... which in this case would be comparing all the properties, yes.
My issue is just that with this def, we could have `wedgeA < wedgeB` and `wedgeB < wedgeA` with `wedgeA != wedgeB` ... which is by itself fine code-wise, but not a good idea nonetheless in my opinion